### PR TITLE
Updated interface for dependency translation rules. No more duplicate…

### PR DESCRIPTION
**Pull request by sorig**
… imports.

Also adds documentation for translation rules.

### CODE REVIEW

File: examples/meta/generator/translate.py [hunk 1]
```diff
@@ -209,67 +209,45 @@ def dependenciesString(self, allClasses, interfacedClasses, enums):
...
+        elif self.targetDict["Dependencies"].get("IncludeInterfacedClasses"):
+            dependencies = dependencies.union(interfacedClasses)
+        elif self.targetDict["Dependencies"].get("IncludeEnums"):
+            dependencies = dependencies.union(enums)
 
-        # Elements are just formatted as their names as default
-        elementTemplate = Template("$className")
-        if "DependencyListElementClass" in self.targetDict["Dependencies"]:
-            elementTemplate = Template(self.targetDict["Dependencies"]["DependencyListElementClass"])
+        translations = set(map(self.translateDependencyElement, dependencies))
```
Comment (sorig) [inline examples/meta/generator/translate.py]: @karlnapf Here I'm making sure that each translated dependency is unique. I.e. if two different dependencies (an enum and a class) result in the same import translation, then they're squashed.

File: examples/meta/generator/targets/doc.js [hunk 2]
```diff
@@ -0,0 +1,239 @@
+/** Target files specify rules for translation from meta-language to a target
```
Comment (karlnapf) [inline examples/meta/generator/targets/doc.js]: what type of file is this?
.js ?

**Comment by karlnapf** [comment]
Great!
Lets merge once travis is green

File: examples/meta/generator/targets/doc.js (hunk 2 above)
Comment (sorig) [inline examples/meta/generator/targets/doc.js]: JavaScript. JSON does not support comments, so syntax highlighters make this hard to read. I could also change to markdown and wrap this in a code segment?

Comment (karlnapf) [inline examples/meta/generator/targets/doc.js]: Lets discuss tomorrow

**Pull request merged**